src/dispatch/plugins/dispatch_slack/service.py

```diff
@@ -22,6 +21,24 @@
 log = logging.getLogger(__name__)
 
 
+class SlackRetryException(Exception):
+    def __init__(self, wait_time):
+        self.wait_time = wait_time
+        super().__init__(f"Retrying slack call in {wait_time} seconds.")
+
+    def get_wait_time(self):
+        return self.wait_time
+
+def slack_wait_strategy(retry_state: RetryCallState):
+    """Determines the wait time for the Slack retry strategy"""
+    exc = retry_state.outcome.exception()
+    if isinstance(exc, SlackRetryException):
+        return exc.get_wait_time()
+
+    # Fallback to exponential backoff if no custom wait time is specified
+    return wait_exponential(multiplier=1, min=1, max=60)(retry_state)
```

Comment:
Ah, yeah, that's true. The better check here would be to look for `get_wait_time` instead. I think my initial thinking was to allow the exception to choose but then default to exponential (instead of 5s).

Reply:
Made some adjustments.